pkg/process/checks/process.go

```diff
@@ -110,14 +131,13 @@ func (p *ProcessCheck) Run(cfg *config.AgentConfig, groupID int32) ([]model.Mess
 	}
 
 	// stores lastPIDs to be used by RTProcess
-	lastPIDs := make([]int32, 0, len(procs))
+	p.lastPIDs = make([]int32, 0, len(procs))
```

Comment:
nit: since everything is synchronous, we could use `p.lastPIDs = p.lastPIDs[:0]` to avoid allocating new memory.